kyaml/openapi/openapi.go

```diff
@@ -695,6 +718,9 @@ func findNamespaceability(paths *spec.Paths) {
 }
 
 func resolve(root interface{}, ref *spec.Ref) (*spec.Schema, error) {
+	if s, ok := root.(*spec.Schema); ok && s == nil {
+		return nil, nil
+	}
```

Comment:
When does this happen?

Reply:
TBH I have no idea... I put breakpoints in all the places I changed the code and ran the tests that hit this issue, and it didn't hit any of my breakpoints. FWIW this is only ever hit in the `setters` code where there are a lot of tests and we have deprecated all setters-related commands. 

I'll see if I can narrow down the issue further though, so that we can be confident that this is safe.

Reply:
TLDR I did a lot of digging and I'm convinced that adding this check here to `resolve` is preserving current behavior. 

Here is what I've discovered:

This panic happens in `TestFixSettersV1` when the SubstitutionCreator is Created and makes a call to validateSubstitutionInfo: 

https://github.com/kubernetes-sigs/kustomize/blob/a464ed0c594bc67a1254855d32485d97092f8555/kyaml/setters2/settersutil/substitutioncreator.go#L64-L67

`validateSubstitutionInfo` makes several checks, one of which is to make a call to `openapi.Resolve`: 

https://github.com/kubernetes-sigs/kustomize/blob/a464ed0c594bc67a1254855d32485d97092f8555/kyaml/setters2/settersutil/substitutioncreator.go#L446-L450

But this panics, because c.SettersSchema isn't set yet, it gets set further down in `SubstitutionCreator.Create`. Moving setting c.SettersSchema doesn't work, because then in some test cases it starts throwing errors it wasn't previously throwing. But I was curious how did this _ever_ work, and I noticed that in the go.sum files, `go-openapi/jsonpointer` is being bumped from v0.19.3 to v0.19.5.

I confirmed this theory by adding a replace directive `replace github.com/go-openapi/jsonpointer => github.com/go-openapi/jsonpointer v0.19.3` to kyaml/go.mod. With this replace directive, the panicking test now passes.

I then hypothesized that previously, if `root` was `nil`, the call to `jsonpointer.Get` silently ignored it, whereas now it panics. I've done some deeper digging as well, and discovered that here, in validationSubstitutionInfo, we are ignoring the returned error:

https://github.com/kubernetes-sigs/kustomize/blob/a464ed0c594bc67a1254855d32485d97092f8555/kyaml/setters2/settersutil/substitutioncreator.go#L446-L450

After some further debugging, I confirmed my theory. When using go-openapi/jsonpointer v0.19.3 and `root` was nil in our `resolve` function, we previously just ignored the error. Now, with go-openapi/jsonpointer v0.19.5, a nil `root` panics. So my check here just checks if `root` is nil, and preserves the previous behavior of ignoring the error and continuing with the other checks.

Reply:
Great investigation, and thank you for the thorough write up!